src/events.rs

```diff
@@ -110,10 +113,25 @@ pub enum WindowEvent {
 pub enum DeviceEvent {
     Added,
     Removed,
-    /// Mouse devices yield `Motion` events where axis `0` is horizontal and axis `1` is vertical.
-    /// A positive value means a movement to the right or the bottom, depending on the axis.
-    /// Such events will be sent even if the mouse is in a corner of the screen.
+
+    /// Change in physical position of a pointing device.
+    ///
+    /// This represents raw, unfiltered physical motion. Not to be confused with `WindowEvent::CursorMoved`.
+    MouseMoved {
+        /// (x, y) change in position in unspecified units.
+        ///
+        /// Different devices may use different units.
+        delta: (f64, f64),
+    },
+
+    /// Physical scroll event
+    MouseWheel {
+        delta: MouseScrollDelta,
+    },
+
+    /// Motion on some analog axis. May report data redundant to other, more specific events.
     Motion { axis: AxisId, value: f64 },
```

Comment:
Not sure if I understood this. Does this mean that whether `Motion` events are also emitted for the mouse is platform-dependent?

Reply:
Mouse events should be emitted on all major desktop platforms.  Mouse events will also be accompanied by Motion events from the same device with the same deltas.  Not all Motion events will be accompanied by Mouse events though, as some Motion events won't come from the Mouse.

Reply:
You could improve this comment to clearly note that `Motion` is a fallback event for any motion along axis that are not already handled by `MouseMoved` or `MouseWheel` (since it seems to be what your implementation does).

Reply:
@vberger Oh hm woops, my Windows and OSX implementation of this behaves differently from the X11 implementation provided by @Ralith .  I'm not using the motion events as a fall back, but I'm instead using them to accompany the mouse events.  I'll need to get those behaviors in sync before we merge.

Reply:
Ah, I see. So, the actual intended behavior is for the generic `Motion` to repeat `MouseMoved`? Is there a clear reason for this repetition?

Reply:
`Motion` is good for being abstract over axes, so it makes sense to include mouse axes in that abstraction.  `MouseMoved` is for when you don't want to be abstract over axes, and you just want the mouse.

Reply:
Would be nice to include a detailed description of this in the docs given the confusion in these reviews.